zkSNACKs/WalletWasabi
# Improve Start WW on Startup feature

**Previously:**
If the user sets the `Start Wasabi with OS start up` option to `true`, then closes WW and deletes the `Registry Key` OR the `.desktop` file OR the `Wasabi Wallet Login Item` (depends on the OS the user has) and after that he launches WW, the option will stay `true`, while WW won't launch. So the value of the option is misleading.

**With this PR**:
We check the `RegKey`/`.desktop file`/`Login Items` before we show the `General Settings` tab to the user, so we can show the user the accurate value. (WW will launch on startup or not)

**How to test**:

1. Get this PR
2. Set the `Start Wasabi with OS start up` to `true`
3. Close WW and delete the  `RegKey`/`.desktop file`/`Login Item`
4. See that the `Setting` and the `UiConfig File` gives you back the real value `(false)`

- [x] Test on Ubuntu 20.14
- [x] Test on macOS

_Might be an edge case, but if it prevents future Github issues, then IMO it worths it._
